src/main/java/uk/nhs/hee/tis/revalidation/migration/util/GmcHexStringConverter.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright 2021 Crown Copyright (Health Education England)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.revalidation.migration.util;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class GmcHexStringConverter {
+  /**
+   * Checks if gmc number has a leading 0 and if so converts to decimal string.
+   * @param gmcNumber gmc number string
+   * @return decimal gmc number
+   */
+  public String convertGmcString(String gmcNumber) {
+    Character leadingChar = gmcNumber.charAt(0);
+    if (leadingChar.equals('0')) {
+      int decimal = Integer.parseInt(gmcNumber, 16);
+      return Integer.toString(decimal);
+    }
```

Comment:
Is this something that is universally true, or just for our current data set?
Looking at the definition a GMC number **could** start with zero, though I'm not aware of any examples.

This example might cover those edge cases better?
```
  @ParameterizedTest
  @ValueSource(strings = {"7654235", "074CB5B"})
  void shouldConvertGmcNumber(String originalGmcNumber) {
    String gmcNumber = Integer.valueOf(originalGmcNumber, 16).toString();

    if (gmcNumber.length() != 7) {
      gmcNumber = originalGmcNumber;
    }
    assertThat("Unexpected gmc number.", gmcNumber, is("7654235"));
  }
```

Reply:
Good question, it was an optimistic assumption that real gmc numbers would not start with a 0, unfortunately as the strings are all 7 digits this was the only way of differentiating between the hex ids and the non-hex ones 

(i.e. 0123456 and 1234567 are both valid hex numbers, but a 7 digit decimal number can at most only ever give a 6 digit hex number)

Reply:
Thinking about it, my example fails for leading character = 0 too.
It's an interesting problem.